File: Data Structures & Algorithms/validate-parentheses/submission-0.java
class Solution {
    public boolean isValid(String s) {
        Stack<Character> stk = new Stack<>();
        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            if (c == '(' || c == '{' || c == '[') {
                stk.push(c);
            } else if (c == ')' && (stk.isEmpty() || stk.pop().charValue() != '(')) return false;
            else if (c == ']' && (stk.isEmpty() || stk.pop().charValue() != '[')) return false;
            else if (c == '}' && (stk.isEmpty() || stk.pop().charValue() != '{')) return false; 
        }
        return stk.isEmpty();
    }
}

File: Data Structures & Algorithms/daily-temperatures/submission-0.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        answer = [0] * len(temperatures)
        stk = []
        for i in range(len(temperatures)):
            while bool(stk) and temperatures[i] > temperatures[stk[-1]]:
                j = stk.pop()
                answer[j] = i - j
            stk.append(i)
        return answer